### ISSUE TITLE
pdfMake - document cannot be generated if it has at least one image inside

### ISSUE THREAD

**Issue by hok89** [bpampuch/pdfmake#710]
Hi,

I have experienced an issue with creation of document on iOS 10 and Safari 10 in Mac OS . In my case the images are causing the problem. I cannot longer add image to a pdf document. 

**Description:**
 Image is encoded as Base64.

 When I am trying to generate document using methods:
-getBuffer
-getBase64
-getDataUrl
-download

I get the same error over and over:
**RangeError: Range consisting of offset and length are out of bounds**

It doesn't matter if image is located in footer/ header/ or simply in content of the file.

If in document there is **no images** everything is fine.
Issue can be reproduce using device with iOS 10 or simply upgrade the Safari desktop to Safari 10 (available in AppStore) .

Thank you.

**Comment by jimothyhalpert7** [comment]
We're experiencing the same issue. However, on OS X, Safari 10 it works fine, and the image gets included. #709 seems to be related to this, but the solution of using https://github.com/pdfmake/pdfmake didn't work for us. Any help would be gladly appreciated, thanks.

**Comment by jimothyhalpert7** [comment]
Switching to `0.1.31` from https://github.com/pdfmake/pdfmake as recommended in  #709 , or to `0.1.18` of this repo worked for us.

**Comment by Akta3d** [comment]
Ok, after search difference I find the problem.
The problem is with the buffer use.

Here all modification I have made in my current version to have image on all browser : 


```
//replace :
/* WEBPACK VAR INJECTION */(function(Buffer, global) {/*!
//by
 /* WEBPACK VAR INJECTION */(function(Buffer) {/*!
```



```
//replace :
Buffer.TYPED_ARRAY_SUPPORT = global.TYPED_ARRAY_SUPPORT !== undefined
             ? global.TYPED_ARRAY_SUPPORT
             : typedArraySupport()
             function typedArraySupport () {
             function Bar () {}
             try {
             var arr = new Uint8Array(1)
             arr.foo = function () { return 42 }
             arr.constructor = Bar
             return arr.foo() === 42 && // typed array instances can be augmented
             arr.constructor === Bar && // constructor can be set
             typeof arr.subarray === 'function' && // chrome 9-10 lack subarray
             arr.subarray(1, 1).byteLength === 0 // ie10 has broken subarray
             } catch (e) {
             return false
             }
             }
//by
Buffer.TYPED_ARRAY_SUPPORT = (function () {
                function Bar () {}
                try {
                    var arr = new Uint8Array(1)
                    arr.foo = function () { return 42 }
                    arr.constructor = Bar
                    return arr.foo() === 42 && // typed array instances can be augmented
                        arr.constructor === Bar && // constructor can be set
                        typeof arr.subarray === 'function' && // chrome 9-10 lack `subarray`
                        arr.subarray(1, 1).byteLength === 0 // ie10 has broken `subarray`
                } catch (e) {
                    return false
                }
            })()
```



```
//comment :
if (Buffer.TYPED_ARRAY_SUPPORT) {
Buffer.prototype.__proto__ = Uint8Array.prototype
Buffer.__proto__ = Uint8Array
}
```


```
//comment :
that.__proto__ = Buffer.prototype
```



```
//replace all:
this[offset] = (value & 0xff)
//by
this[offset] = value

//same with [offset + 1], [offset + 3]
```



```
//replace :
/* WEBPACK VAR INJECTION */ }.call(exports, __webpack_require__(2).Buffer, (function() { return this; }())))
//by
/* WEBPACK VAR INJECTION */}.call(exports, __webpack_require__(2).Buffer))
```

**Comment by liborm85** [comment]
Solve this problem, the latest released version (0.1.22) ?
